src/jit/emit.cpp

```diff
@@ -5189,20 +5188,27 @@ UNATIVE_OFFSET emitter::emitDataGenBeg(UNATIVE_OFFSET size, bool dblAlign, bool
 
     secOffs = emitConsDsc.dsdOffs;
 
-    /* Are we require to align this request on an eight byte boundry? */
-    if (dblAlign && (secOffs % sizeof(double) != 0))
+    if (align)
     {
-        /* Need to skip 4 bytes to honor dblAlign */
-        /* Must allocate a dummy 4 byte integer */
-        int zero = 0;
-        emitDataGenBeg(4, false, false);
-        emitDataGenData(0, &zero, 4);
-        emitDataGenEnd();
+        assert(size <= 16);
```

Comment:
Wouldn't this assert hold for any case (not just when `align` is true)?

Reply:
Constants can have any size (e.g. 32 bytes for AVX stuff), it's just that alignment is currently supported only for 4, 8 and 16 bytes constants. Hmm, maybe I should check those specific sizes instead

Reply:
I added a comment to clarify this assert.